src/Controllers/ElementalAreaController.php

```diff
@@ -29,23 +33,217 @@ class ElementalAreaController extends CMSMain
+            $area->Elements()->add($clone);
+            // reorder
+            $reorderer = Injector::inst()->create(ReorderElements::class, $clone);
+            $reorderer->reorder($id);
+            return $this->jsonSuccess(201);
+        } catch (Exception) {
+            $this->jsonError(500);
+        }
+    }
+
+    public function apiPublish(HTTPRequest $request): HTTPResponse
+    {
+        $this->checkJsonCsrfToken($request);
+        $id = $this->getPostData()['id'] ?? '';
+        $element = BaseElement::get()->byID($id);
+        if (!$element) {
+            $this->jsonError(400);
+        }
+        if (!$element->canPublish()) {
+            $this->jsonError(403);
+        }
+        $element->publishRecursive();
+        return $this->jsonSuccess(204);
+    }
+
+    public function apiReadElementalArea(HTTPRequest $request): HTTPResponse
+    {
+        $request = $this->getRequest();
+        $elementalAreaID = $request->param('elementalAreaID');
+        $elementalArea = ElementalArea::get()->byID($elementalAreaID);
+        if (!$elementalArea) {
+            $this->jsonError(404);
+        }
+        if (!$elementalArea->canView()) {
+            $this->jsonError(403);
+        }
+        $data = [];
+        foreach ($elementalArea->Elements() as $element) {
+            if (!$element->canView()) {
+                continue;
+            }
+            $data[] = [
+                'id' => $element->ID,
+                'title' => $element->Title,
+                'blockSchema' => $element->getBlockSchema(),
+                'obsoleteClassName' => $element->getObsoleteClassName(),
+                'version' => $element->Version,
+                'isPublished' => $element->isPublished(),
+                'isLiveVersion' => $element->isLiveVersion(),
+                'canDelete' => $element->canDelete(),
+                'canPublish' => $element->canPublish(),
+                'canUnpublish' => $element->canUnpublish(),
+                'canCreate' => $element->canCreate(),
+            ];
+        }
+        $this->extend('updateApiReadElementalArea', $data, $request);
+        return $this->jsonSuccess(200, $data);
+    }
+
+    public function apiSort(HTTPRequest $request): HTTPResponse
+    {
+        $this->checkJsonCsrfToken($request);
+        $postData = $this->getPostData();
+        $id = $postData['id'] ?? 0;
+        $afterBlockID = $postData['afterBlockID'] ?? 0;
+        $element = BaseElement::get()->byID($id);
+        if (!$element) {
+            $this->jsonError(400);
+        }
+        if (!$element->canEdit()) {
+            $this->jsonError(403);
+        }
+        try {
+            $reorderingService = Injector::inst()->create(ReorderElements::class, $element);
+            $reorderingService->reorder($afterBlockID);
+            return $this->jsonSuccess(204);
+        } catch (Exception) {
+            $this->jsonError(500);
+        }
```

Comment:
Shouldn't that just happen anyway? Why do we need to explicitly catch and throw like this?
If this was needed I'd expect to see it in literally all of our actions on all of our controllers.

Reply:
Removed the try block, it doesn't need to be there. It'll 500 on any exception thrown anyway

Reply:
My point is more that a 500 is a 500. And an exception will always result in a 500.

If we are catching exceptions _at all_ and using `$this->jsonError(500);` in the catch block, then we should do this _everywhere_ we want a json response, because anything could throw an exception we're not expecting.

And we _shouldn't_ do that everywhere because it should be handled globally instead. Which it is already, isn't is?

Reply:
That's try, have removed